File: 2-Ad-Hoc/1103-Alarme-Despertador/1103.py
while True:
    h = input().split()
    if h[0] == '0' and h[1] == '0' and h[2] == '0' and h[3] == '0':
        break
    if h[0] == '0':
        h[0] = '24'
    if h[2] == '0':
        h[2]  = '24'
    h = [int(i) for i in h]
    if h[0] < h[2] or h[0] == h[2] and h[1] <= h[3]:
        print (int((h[2] * 60 + h[3]) - (h[0] * 60 + h[1])))
    else:
        print (int((h[2] * 60 + 24 * 60 + h[3]) - (h[0] * 60 + h[1])))

File: 2-ad-hoc/1103.cpp
#include <iostream>
#include <cstring>
#include <vector>
// 
int main(){
    int epa,tot,h_pas,m_pas;
    std::string inp;
    std::vector<std::vector<int>> valor;
    char *split;
    while(true){
        epa=0;
        tot=0;
        std::getline(std::cin,inp);
        split=std::strtok(&inp[0]," ");
        valor.push_back({});
        for(int i=0;split!=NULL;++i){
            valor[valor.size()-1].push_back(std::atoi(split));
            if(*split=='0'){
                ++epa;
            }
            if(valor[valor.size()-1].size()>1){
                valor.push_back({});
            }
            split=std::strtok(NULL," ");
        }
        if(epa>3){
            break;
        }
        h_pas=valor[1][0]-valor[0][0];
        (h_pas<0)?h_pas+=23:h_pas;
        m_pas=valor[1][1]-valor[0][1];
        if(m_pas<0){
            m_pas+=60;
            --h_pas;
        }
        std::cout<<h_pas*60+m_pas<<std::endl;
        valor.clear();
        inp.clear();
    }
    return 0;
}